packages/shared/lib/services/sync/config/config.service.ts

```diff
@@ -487,43 +487,33 @@ export async function getSyncConfigsWithConnectionsByEnvironmentId(environment_i
     return result;
 }
 
-export async function getSyncConfigsWithConnections(
-    providerConfigKey: string,
-    environment_id: number
-): Promise<{ connections: { connection_id: string }[]; provider: string; unique_key: string }[]> {
-    const result = await db.readOnly
-        .select(
-            `${TABLE}.id`,
-            '_nango_configs.provider',
-            '_nango_configs.unique_key',
-            db.knex.raw(
-                `(
-                    SELECT json_agg(
-                        json_build_object(
-                            'connection_id', _nango_connections.connection_id
-                        )
-                    )
-                    FROM _nango_connections
-                    WHERE _nango_configs.environment_id = _nango_connections.environment_id
-                    AND _nango_configs.unique_key = _nango_connections.provider_config_key
-                    AND _nango_configs.deleted = false
-                    AND _nango_connections.deleted = false
-                ) as connections
-                `
-            )
+export async function getConnectionCountsByProviderConfigKey(
+    environmentId: number
+): Promise<{ data: { providerConfigKey: string; total: string }[]; total: number }> {
+    const q = db.knex
+        .select<{ providerConfigKey: string; total: string }[]>(
+            '_nango_configs.unique_key as providerConfigKey',
+            db.knex.raw(`
+                COUNT(DISTINCT CASE WHEN _nango_sync_configs.active = true AND _nango_sync_configs.deleted = false THEN _nango_connections.id END) as connectionsWithScripts
+            `),
+            db.knex.raw(`
+                COUNT(DISTINCT _nango_connections.id) as total
+            `)
         )
-        .from<DBSyncConfig>(TABLE)
-        .join('_nango_configs', `${TABLE}.nango_config_id`, '_nango_configs.id')
-        .where({
-            '_nango_configs.environment_id': environment_id,
-            '_nango_configs.unique_key': providerConfigKey,
-            active: true,
-            enabled: true,
-            '_nango_configs.deleted': false,
-            [`${TABLE}.deleted`]: false
-        });
-
-    return result;
+        .from('_nango_connections')
+        .join('_nango_configs', '_nango_connections.provider_config_key', '_nango_configs.unique_key')
+        .leftJoin('_nango_sync_configs', `_nango_sync_configs.nango_config_id`, '_nango_configs.id')
+        .where('_nango_connections.environment_id', environmentId)
+        .andWhere('_nango_connections.deleted', false)
+        .andWhere('_nango_configs.deleted', false)
+        .groupByRaw('ROLLUP("_nango_configs"."unique_key")');
```

Comment:
I am not sure what the ROLLUP with a single column does? Can you explain?

Reply:
Rollup will summarize the column. The query count per unique_key, and then the last row `unique_key` will be null and the count will be the total. I could do it by hand tbh, it's just one less loop.
![Screenshot 2025-04-23 at 17 11 10](https://github.com/user-attachments/assets/60ce784e-1e4e-4919-b9fc-bc518cf0e3e0)